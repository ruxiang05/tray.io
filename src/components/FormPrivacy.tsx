import React from "react";
import FormInput from "./FormInput";
import { FormProps } from "./Form";
import SubmitButton from "./SubmitButton";

const Privacy = ({ changePage, handleChange }: FormProps) => (
  <div>
    <FormInput
      type="checkbox"
      name="productComm"
      text="Receive updates about Tray.io product by email"
      handleChange={handleChange}
    />
    <FormInput
      type="checkbox"
      name="otherComm"
      text="Receive communication by email for other products created by the Tray.io
      team"
      handleChange={handleChange}
    />
    <SubmitButton changePage={() => changePage("done")} />
  </div>
);

export default Privacy;
