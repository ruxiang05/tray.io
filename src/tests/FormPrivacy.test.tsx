import React from "react";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";
import FormPrivacy from "../components/FormPrivacy";

describe("FormPrivacy", () => {
  const mock = {
    changePage: jest.fn(),
    handleChange: jest.fn(),
  };

  it("should render", () => {
    const tree = renderer.create(<FormPrivacy {...mock} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
