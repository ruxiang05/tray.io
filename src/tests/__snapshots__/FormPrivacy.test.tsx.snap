// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormPrivacy should render 1`] = `
<div>
  <div
    className="form-input"
  >
    <label
      htmlFor="productComm"
    >
      Receive updates about Tray.io product by email
    </label>
    <input
      name="productComm"
      onChange={[MockFunction]}
      type="checkbox"
    />
  </div>
  <div
    className="form-input"
  >
    <label
      htmlFor="otherComm"
    >
      Receive communication by email for other products created by the Tray.io team
    </label>
    <input
      name="otherComm"
      onChange={[MockFunction]}
      type="checkbox"
    />
  </div>
  <button
    onClick={[Function]}
    type="button"
  >
    Submit
  </button>
</div>
`;
